dandi/dandiarchive.py

```diff
@@ -1,19 +1,13 @@
-"""
-Interactions with DANDI archive ATM go either through Girder or through DANDI API.
-
-This module provides helpers to decide on which API Client to use and/or
-abstract higher level code from specifics of any particular client.
-
-Eventually it is largely to be "dissolved" whenever we stop talking to girder.
-"""
-
+from abc import ABC, abstractmethod
 from contextlib import contextmanager
 import re
+from typing import Iterator, Optional, Tuple
 from urllib.parse import unquote as urlunquote
 
+from pydantic import AnyHttpUrl, BaseModel, parse_obj_as, validator
```

Comment:
just a note: likely because of this module RF and extra imports (but didn't investigate), code of this PR adds ~40 msec to `dandi --help` (to the total of 800ms, so about 5%)... at some point we might review import paths etc if we keep growing - since waiting seconds for `--help` might be a bit too much, and would be impediment if we add some shell completions support (IIRC I did not find any ready to use for `click` such as argcomplete which was odd)

Reply:
FYI, click has command completion built-in: https://click.palletsprojects.com/en/7.x/bashcomplete/ (though I think some details may be changing in Click 8).

Reply:
cool, filed https://github.com/dandi/dandi-cli/issues/608 . Feel welcome to self-assign but it is of no urgency so I didn't